src/cmd/singularity/cli/build_linux.go

```diff
@@ -42,9 +41,9 @@ func run(cmd *cobra.Command, args []string) {
 			sylog.Fatalf("Unable to submit build job: %v", authWarning)
 		}
 
-		def, err := types.NewDefinitionFromURI(spec)
+		def, err := definitionFromSpec(spec)
 		if err != nil {
-			return
```

Comment:
This return is why you were getting silent failures - no debug output, or anything at all, just done.